### DISCUSSION TITLE
running start.R asks for various package installations every time

### DISCUSSION THREAD

**Discussion by sreyam-sengupta** [magpiemodel/magpie#615] [Help] [answered]
Dear all,

Thanks for reading!

**Error background and description:**
When I try to run source("start.R"), either from R shell or from Rstudio IDE, it asks me to download and install a number of packages (screenshots attached). This happens every time. When I press Y(es), the packages are downloaded and installed. Previously, after this step, it displayed a menu containing various start scripts and their descriptions, i.e. things were running normally.

But now, I get the following error message:
```
Error: package or namespace load failed for ‘lucode2’ in library.dynam(lib, package, package.lib):
 DLL ‘data_table’ not found: maybe not installed for this architecture?
```
this, inspite of having downloaded and installed lucode2 multiple times from source, with all dependencies. I'm not sure what's going on, and any help would be appreciated.

**System and version specifications:**
OS: MS Windows 10 Pro
System architecture: x64
R version 4.3.2
renv 1.0.3
GAMS version 45
MAgPIE branch: https://github.com/jansteinhauser/magpie.git (builds on version 4.6.7)

Please let me know if there's any other relevant info I can provide. Thanks!

![page_1](https://github.com/magpiemodel/magpie/assets/36034938/cc47265d-a7dd-4c64-9ca2-fd322d355f17)
![page_2](https://github.com/magpiemodel/magpie/assets/36034938/a8d02842-a2f5-4849-9531-e3c8d55bba3c)
![page_3](https://github.com/magpiemodel/magpie/assets/36034938/567bd24f-5c46-45d1-aabf-9a0b24aaed7d)

**Comment by pfuehrlich-pik** [comment, marked as answer]
Seems like the bootstrapping process is not happening as planned. Only the first time when you start an R session in the magpie folder (this is crucial) your renv will be filled with all the packages you need, because the [.Rprofile](https://github.com/magpiemodel/magpie/blob/master/.Rprofile) is run. After that you should never run into the requirement check again that asks you to install everything. So I'd suggest you clone magpie from scratch, and then start an R session in that folder. You should get the message "R package dependencies are not installed in this renv, installing now...". If you don't or errors/warnings come up feel free to post them here.

**Reply by sreyam-sengupta** [reply]
Thanks a lot, I'll try that!

**Reply by sreyam-sengupta** [reply]
I solved the issue: I had originally cloned the repo to a network drive (local to my institute, but not to my PC). That was the source of error. I deleted that repo and re-cloned a fresh instance to the C drive (local to my PC). Now everything works perfectly, thanks!
